docs/developer/crd-versioning.md

```diff
@@ -0,0 +1,46 @@
+# CRD Versioning
+
+This document discusses how we version our CustomResourceDefinitions (CRDs) that define Policies, Filters, or other configuration objects. NGINX Gateway Fabric distributes a few CRDs, such as NginxGateway, NginxProxy, ClientSettingsPolicy, and ObservabilityPolicy. More will inevitably be added over time.
+
+## Initial Version
+
+When creating a new CRD, the initial version should be `v1alpha1`. This version indicates a new API that needs time for use and feedback. By starting at the alpha version, it also allows us a little more flexibility in changing the API.
+
+## Changing an API
+
+When changing an API in a CRD, it's important to understand the impact of those changes. This will determine if and how we need to update the API version.
+
+**No change in version required**
+
+- Adding new optional fields.
+
+**Compatible changes, requires a version change**
```

Comment:
Can we add a link to the k8s doc on compatible changes somewhere in this section: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#on-compatibility?